From 809bf8ec6f532c9aae5c280ede095f3fb095cfdf Mon Sep 17 00:00:00 2001
From: Alain Volmat <alain.volmat@foss.st.com>
Date: Fri, 8 Nov 2024 15:21:39 +0100
Subject: [PATCH] media: stm32: dcmipp: fix advertise aux pixelproc sink
 formats

Remove 2 duplicate YUV422 2X8 formats and only keep UYVY8_1X16
for csi input since that's the only one supported by the DCMIPP
to ensure correct behavior of the whole pipe.
Regarding parallel interface based pixelproc input, the pixel pipe
needs a YUYV8_2X8, aka 4 cycles, Y / U / Y / V hence removing the
other formats as input.

Change-Id: If40b96640a28ae4a39b9bd50a82de66633c9ed18
Signed-off-by: Alain Volmat <alain.volmat@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/422108
Reviewed-by: Philippe CORNU <philippe.cornu@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Alain VOLMAT <alain.volmat@st.com>
Reviewed-by: Alain VOLMAT <alain.volmat@st.com>
Domain-Review: Philippe CORNU <philippe.cornu@foss.st.com>
---
 .../platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c     | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
index e9fa98e1db24..9bd891645e34 100644
--- a/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
+++ b/drivers/media/platform/st/stm32/stm32-dcmipp/dcmipp-pixelproc.c
@@ -121,15 +121,7 @@ static const struct dcmipp_pixelproc_pix_map dcmipp_pixelproc_sink_pix_map_list[
 	PIXMAP_MBUS(RGB888_3X8, DCMIPP_AUX_PIPE),
 	/* YUV formats */
 	PIXMAP_MBUS(YUYV8_2X8, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(YUYV8_1X16, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(YUYV8_2X8, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(UYVY8_2X8, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(UYVY8_1X16, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(UYVY8_2X8, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(YVYU8_2X8, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(YVYU8_1X16, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(VYUY8_2X8, DCMIPP_AUX_PIPE),
-	PIXMAP_MBUS(VYUY8_1X16, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(Y8_1X8, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(Y10_1X10, DCMIPP_AUX_PIPE),
 	PIXMAP_MBUS(Y12_1X12, DCMIPP_AUX_PIPE),
-- 
2.47.2

